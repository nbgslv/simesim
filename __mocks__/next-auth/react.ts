const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
