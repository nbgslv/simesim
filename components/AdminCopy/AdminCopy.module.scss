@import "styles/variables";

.main {
  :global .btn path {
    color: map-get($theme-colors, 'primary') !important;
  }
}

.copyButton {
  margin: 0;
  padding: 0;
  background: transparent;
  border: none;
  font-size: .875rem;
  display: flex;
  justify-content: center;

  &:hover {
    background: transparent;
    border: none;
  }
}