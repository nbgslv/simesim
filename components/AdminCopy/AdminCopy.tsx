import React, { useRef, useState } from 'react';
import { Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { regular } from '@fortawesome/fontawesome-svg-core/import.macro';
import { toast } from 'react-toastify';
import { AnimatePresence, motion } from 'framer-motion';
import styles from './AdminCopy.module.scss';
import ToastContent from '../Toast/ToastContent';

const AdminCopy = ({ children }: { children: React.ReactNode }) => {
  const [showCopy, setShowCopy] = useState(false);
  const copySpanRef = useRef<HTMLSpanElement>(null);

  const handleCouponCodeCopy = () => {
    if (copySpanRef.current) {
      // Copy the text inside the text field
      navigator.clipboard.writeText(copySpanRef.current.innerText);
      toast.success(
        <ToastContent
          content={`${copySpanRef.current.innerText.substring(
            0,
            10
          )}... was copied successfully`}
        />
      );
    }
  };

  return (
    <div className={styles.main}>
      <Button
        onMouseEnter={() => setShowCopy(true)}
        onMouseLeave={() => setShowCopy(false)}
        onClick={handleCouponCodeCopy}
        className={styles.copyButton}
      >
        <span className="me-1" ref={copySpanRef}>
          {children}
        </span>
        <AnimatePresence initial={false}>
          {showCopy && (
            <motion.div
              initial={{ opacity: 0 }}
              animate={{ opacity: 1 }}
              exit={{ opacity: 0 }}
            >
              <FontAwesomeIcon icon={regular('copy')} />
            </motion.div>
          )}
        </AnimatePresence>
      </Button>
    </div>
  );
};

export default AdminCopy;
