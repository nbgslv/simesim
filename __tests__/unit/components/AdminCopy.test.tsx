import React from 'react';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { ToastContainer } from 'react-toastify';
import AdminCopy from '../../../components/AdminCopy/AdminCopy';

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

describe('AdminCopy', () => {
  it('renders the copy button', () => {
    render(<AdminCopy value="Text to copy" />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('copies the text to clipboard when the copy button is clicked', async () => {
    jest.spyOn(navigator.clipboard, 'writeText');
    render(<AdminCopy value="Text to copy" />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(navigator.clipboard.writeText).toHaveBeenCalledWith('Text to copy');
  });

  it('shows a toast to confirm the text was copied to clipboard', async () => {
    // jest.spyOn(navigator.clipboard, 'writeText');
    jest.useFakeTimers();
    render(
      <>
        <AdminCopy value="Text to copy" />
        <ToastContainer />
      </>
    );
    const button = screen.getByRole('button');
    fireEvent.click(button);
    act(() => {
      jest.advanceTimersByTime(1000);
    });
    expect(
      screen.getAllByText(/was copied successfully/).length
    ).toBeGreaterThanOrEqual(1);
  });
});
