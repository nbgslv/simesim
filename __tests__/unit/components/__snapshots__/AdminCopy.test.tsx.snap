// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminCopy matches snapshot 1`] = `
<div>
  <div
    class="main"
  >
    <button
      class="copyButton btn btn-primary"
      type="button"
    >
      <span
        class="me-1"
      >
        Text to copy
      </span>
    </button>
  </div>
</div>
`;
